packages/graphql-server-core/src/runHttpQuery.ts

```diff
@@ -68,6 +70,13 @@ export async function runHttpQuery(handlerArguments: Array<any>, request: HttpQu
 
   let responses: Array<ExecutionResult> = [];
   for (let requestParams of requestPayload) {
+    if ( isGetRequest && !requestParams.query.trim().startsWith('query')) {
```

Comment:
I would check for "document doesn't contain the keyword `mutation` outside of all brackets"